src/test/java/com/jitterted/mobreg/adapter/in/web/member/EnsembleSummaryViewTest.java

```diff
@@ -10,108 +13,172 @@
+            .withGithubUsername("participant_username")
+            .buildAndSave();
+        ensemble.acceptedBy(member.getId());
 
         EnsembleSummaryView ensembleSummaryView = EnsembleSummaryView
-                .toView(ensemble, memberId);
+            .toView(ensemble, member.getId(), memberBuilder.memberService());
 
         assertThat(ensembleSummaryView.memberStatus())
-                .isEqualTo("accepted");
+            .isEqualTo("accepted");
     }
 
     @Test
     public void noRecordingEnsembleThenViewIncludesEmptyLink() throws Exception {
         Ensemble ensemble = EnsembleFactory.withIdOf1AndOneDayInTheFuture();
+        MemberService memberService = new MemberService(new InMemoryMemberRepository());
 
-        EnsembleSummaryView ensembleSummaryView = EnsembleSummaryView.toView(ensemble, MemberId.of(1));
+        EnsembleSummaryView ensembleSummaryView = EnsembleSummaryView.toView(ensemble, MemberId.of(1), memberService);
 
         assertThat(ensembleSummaryView.recordingLink())
-                .isEmpty();
+            .isEmpty();
     }
 
     @Test
     public void ensembleWithRecordingThenViewIncludesStringOfLink() throws Exception {
         Ensemble ensemble = EnsembleFactory.withIdOf1AndOneDayInTheFuture();
         ensemble.linkToRecordingAt(URI.create("https://recording.link/abc123"));
+        MemberService memberService = new MemberService(new InMemoryMemberRepository());
 
-        EnsembleSummaryView ensembleSummaryView = EnsembleSummaryView.toView(ensemble, MemberId.of(1));
+        EnsembleSummaryView ensembleSummaryView = EnsembleSummaryView.toView(ensemble, MemberId.of(1), memberService);
 
         assertThat(ensembleSummaryView.recordingLink())
-                .isEqualTo("https://recording.link/abc123");
+            .isEqualTo("https://recording.link/abc123");
     }
 
     @Test
     public void viewContainsGoogleCalendarLink() throws Exception {
         Ensemble ensemble = EnsembleFactory.withIdOf1AndOneDayInTheFuture();
+        MemberService memberService = new MemberService(new InMemoryMemberRepository());
 
-        EnsembleSummaryView ensembleSummaryView = EnsembleSummaryView.toView(ensemble, MemberId.of(1));
+        EnsembleSummaryView ensembleSummaryView = EnsembleSummaryView.toView(ensemble, MemberId.of(1), memberService);
 
         String expectedLink = new GoogleCalendarLinkCreator().createFor(ensemble);
         assertThat(ensembleSummaryView.googleCalendarLink())
-                .isEqualTo(expectedLink);
+            .isEqualTo(expectedLink);
     }
 
     @Test
     public void viewIndicatesNotAbleToAcceptIfEnsembleIsFullAndCurrentlyUnknown() throws Exception {
         Ensemble ensemble = EnsembleFactory.withIdOf1AndOneDayInTheFuture();
-        MemberFactory.acceptCountMembersWith(ensemble, 5);
+        TestMemberBuilder memberBuilder = new TestMemberBuilder();
+        memberBuilder.createAnsSaveMembers(5);
+        EnsembleFactory.acceptCountMembersFor(5, ensemble);
```

Comment:
This is kinda "yucky". It's making assumptions around the IDs that are assigned across two different creators (builder + factory). And it's doing all this for no reason that the _test_ requires, i.e., the assertion hasn't changed. Perhaps a Stub MemberService is needed?

Reply:
Thanks, that's a great idea. I felt the yuck and wasn't sure what to do about it.